From 725ab072130ca3ce4104f4351e48fe50f57ae330 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Mon, 17 Oct 2022 11:43:30 +0200
Subject: [PATCH 164/167] offer an opt out of automatic xdg-desktop-portal use

this allows users to skip over the portal invocation if it is
undesirable (e.g. during early stages of session setup on KDE's Plasma)

Change-Id: I43e3c3a59b9bb71c77012f8f0ec7ad7ba904a050
Reviewed-by: David Edmundson <davidedmundson@kde.org>
(cherry picked from commit a7bd656cf40b0cf7ee7f96e6c59974804a4894ba)
---
 .../services/genericunix/qgenericunixservices.cpp              | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/platformsupport/services/genericunix/qgenericunixservices.cpp b/src/platformsupport/services/genericunix/qgenericunixservices.cpp
index 210e843e6d..2abe039126 100644
--- a/src/platformsupport/services/genericunix/qgenericunixservices.cpp
+++ b/src/platformsupport/services/genericunix/qgenericunixservices.cpp
@@ -393,6 +393,9 @@ private:
 QGenericUnixServices::QGenericUnixServices()
 {
 #if QT_CONFIG(dbus)
+    if (qEnvironmentVariableIntValue("QT_NO_XDG_DESKTOP_PORTAL") > 0) {
+        return;
+    }
     QDBusMessage message = QDBusMessage::createMethodCall(
             QStringLiteral("org.freedesktop.portal.Desktop"), QStringLiteral("/org/freedesktop/portal/desktop"),
             QStringLiteral("org.freedesktop.DBus.Properties"), QStringLiteral("Get"));
-- 
2.38.1

